.pre-commit-config.yaml

```diff
@@ -0,0 +1,31 @@
+repos:
+    -   repo: https://github.com/pycqa/isort
+        rev: 5.8.0
+        hooks:
+        -    id: isort
+             additional_dependencies: ["colorama>=0.4.3"]
```

Comment:
Do package dependencies not get installed?

Reply:
`pre-commit` runs these hooks in isolated environments so any additional dependencies must be specified so that pre-commit knows to install them before running the hook.

Reply:
For example, 

```
-   repo: https://github.com/pycqa/isort
        rev: 5.8.0
        hooks:
        -    id: isort
             additional_dependencies: ["colorama>=0.4.3"]
```

The above config says install repo from `https://github.com/pycqa/isort` with rev/tag `5.8.0` with the additional dependency of `colorama`

Reply:
How does it install isort then?

Reply:
At a high level,
- Create environment
- Git clone specific tag and install

Ref: [here](https://pre-commit.com/#pre-commit-configyaml---repos)

Reply:
> Git clone specific tag and install

What does "install" mean? If they use `pip install` or `python setup.py install` that would install the dependencies as well.

Reply:
Yes but colorama is an optional dependency. Ref: [here](https://github.com/PyCQA/isort/blob/995e016b9d888fea02bedeb5be8ea7f01b694b58/pyproject.toml#L47)

Reply:
"The hook repository must be installable via pip install . (usually by either setup.py or pyproject.toml). The installed package will provide an executable that will match the entry – usually through console_scripts or scripts in setup.py."

https://pre-commit.com/#python